### ISSUE TITLE
localStorage polyfill for Safari private browsing 

### ISSUE THREAD

**Issue by skiraikou** [polyfillpolyfill/polyfill-service#1202]
localStorage polyfill  doesn't work properly for the Safari private browsing case 
it only check the existance of localStorage but not the localStorage  accessibility ( setItem/getItem doesn't work unless localStorage object exist)

**Comment by JakeChampion** [comment]
@PfxMobile We are happy to accept a PR which improves our feature detection inside the polyfill :-)

**Comment by englishextra** [comment]
```if(!("undefined"!==typeof window.localStorage&&"undefined"!==typeof window.sessionStorage)){loadJS("./cdn/polyfills/0.1/js/Storage.fixed.min.js");}```

?

https://github.com/remy/polyfills/blob/master/Storage.js

**Comment by skiraikou** [comment]
try {
    // Test webstorage existence.
    if (!window.localStorage || !window.sessionStorage) throw "exception";
    // Test webstorage accessibility - Needed for Safari private browsing.
    window.localStorage.setItem('storage_test', 1);
    window.localStorage.removeItem('storage_test');
  } catch (e) {

This is how it's done in https://github.com/AgentME/webstorage-polyfill

**Comment by englishextra** [comment]
@PfxMobile I tested that on win 10 fedora 24 and abdroid 4.3. Cant speak for IOS if I got it right.

**Comment by skiraikou** [comment]
That's really an issue of Safari Private mode - localStorage exist but the setItem/getItem method throws an error

**Comment by JakeChampion** [comment]
@PfxMobile Can you submit that as a pull-request?

**Comment by skiraikou** [comment]
Sure,will  try to provide it this week. Created this just to track an issue in case of someone already interested in the same

**Comment by rubencodes** [comment]
This would be awesome, I'm currently handling this manually. Important to note that Safari is weird, and I'm not sure the existing polyfill would suffice. See: https://github.com/AgentME/webstorage-polyfill/blob/master/index.js#L105

**Comment by JakeChampion** [comment]
PR for this work https://github.com/Financial-Times/polyfill-service/pull/1345

**Comment by citta-lab** [comment]
localStorage polyfill doesn't get detected in Safari 10 (private mode). I think we might need to update config.json to include safari 10 ? 
https://cdn.polyfill.io/v2/polyfill.js?features=localStorage
```
/* Polyfill service v3.25.1
 * For detailed credits and licence information see https://github.com/financial-times/polyfill-service.
 * 
 * UA detected: safari/10.1.0
 * Features requested: localStorage
 *  */
(function(undefined) {
/* No polyfills found for current settings */
})
.call('object' === typeof window && window || 'object' === typeof self && self || 'object' === typeof global && global || {});
```
